algebird-test/src/test/scala/com/twitter/algebird/macros/CaseClassMacrosTest.scala

```diff
@@ -0,0 +1,90 @@
+package com.twitter.algebird.macros
+
+import org.scalatest._
+import com.twitter.algebird._
+import com.twitter.algebird.macros.caseclass._
+
+class CaseClassMacrosTest extends WordSpec with Matchers {
+
+  case class Foo(a: Int, b: Int)
+  case class DoubleFoo(a: Double, b: Double)
+
+  "SemigroupMacro" should {
```

Comment:
there are stronger laws you can use. See:

https://github.com/twitter/algebird/blob/develop/algebird-test/src/test/scala/com/twitter/algebird/CountMinSketchTest.scala#L27

So, we should be able to check `semigroupLaws[Foo]` and we can be sure everything is correct. You will need to implement a generator for Foo (which again, we could implement if no one else has, an Arbitrary[T] macro for case classes).